server/router/services/domain-service/helpers/fetch-domain-list-next-page.spec.js

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2021 Uber Technologies Inc.
+//
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import { delay } from '../../../helpers';
+import fetchDomainListNextPage from './fetch-domain-list-next-page';
+
+jest.mock('../../../helpers', () => ({
+  delay: jest.fn(),
+}));
+
+describe('fetchDomainListNextPage', () => {
+  const getMockContext = domainResponse => {
+    const ctx = {
+      cadence: {
+        listDomains: jest.fn().mockImplementation(() => domainResponse),
+      },
+    };
+
+    return ctx;
+  };
+
+  describe('when no nextPageToken returned', () => {
+    it('should return domainList.', async () => {
+      const domains = ['domain1', 'domain2'];
+      const domainResponse = {
+        domains,
+      };
+
+      const ctx = getMockContext(domainResponse);
+
+      const domainList = await fetchDomainListNextPage({
+        ctx,
+      });
+
+      expect(domainList).toEqual(domains);
+    });
+  });
+
+  describe('when nextPageToken is returned', () => {
```

Comment:
do you need a case where the cache is updated and the result set changed during pagination?

Reply:
I don't think that case can occur in the same request.

so to put another way:

say request1 is using the cached version
following path from

Reply:
cadence-server will continue to paginate even if the result has changed since is my understanding